src/App.tsx

```diff
@@ -2,43 +2,81 @@ import { ChangeEvent, useState } from 'react';
 import styles from './App.module.scss';
 import TasksList from './components/TasksList';
 import NewTask from './components/NewTask';
-
-
+import 'semantic-ui-css/semantic.min.css';
 
 function App() {
-    const [tasksArray, setTasksArray] = useState([{ id: 0, item: "Buy Milk" }, { id: 1, item: "Go Run" }, { id: 2, item: "Make Dinner" }, { id: 3, item: "Bake Cake" }]);
-    const [newTaskItem, setNewTaskItem] = useState('')
+    const [tasksArray, setTasksArray] = useState([
+        { id: 0, item: 'Buy Milk' },
+        { id: 1, item: 'Go Run' },
+        { id: 2, item: 'Make Dinner' },
+        { id: 3, item: 'Bake Cake' },
+    ]);
+    const [newTaskItem, setNewTaskItem] = useState('');
+    const [editedTaskItem, setEditedTaskItem] = useState('');
+    const [editTaskId, setEditTaskId] = useState<number | undefined>();
 
     function textTyping(event: ChangeEvent<HTMLInputElement>) {
-        setNewTaskItem(event.target.value)
+        const { name } = event.target;
+        name === 'new-task-text'
+            ? setNewTaskItem(event.target.value)
+            : setEditedTaskItem(event.target.value);
     }
 
-    function clearTyping(){
-        setNewTaskItem("")
+    function clearTyping() {
+        setNewTaskItem('');
     }
 
```

Comment:
redundant.
You should use form HTML element with submit and clear.

Reply:
Let's talk about it. The reason I'm using it other way because the clear button not only clears the text from the input but disables itself and becomes grayed-out for changes until another value is typed into the inbox again.